scripts/builtin/imputeByKNN.dml

```diff
@@ -0,0 +1,208 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+# Impute the data by using KNN-algorithm and finding the nearest neighbor using euclidean distance
+# Currently only for single column with multiple missing values and top 1 neighbor
+#
+# INPUT:
+# -------------------------------------------------------------------------------------
+# data    Data Matrix (numerical)
+# method  methods of calculating the KNN-algorith depending on the size of data and missing values
+# -------------------------------------------------------------------------------------
+#
+# OUTPUT:
+# -----------------------------------------------------------------------------------
+# result     imputed dataset
+# -----------------------------------------------------------------------------------
+
+
+#method 1
+#impute By Mean, really small, coarse grained-operation
+#Extract Top 1 distances, impute the respective value
+#dist runtime * #features
```

Comment:
this (dist runtime * #features) seems incorrect as the distance computation is already O(N2*#features).